src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs

```diff
@@ -41,7 +42,7 @@ public Task StartAsync(CancellationToken cancellationToken)
                 // This method is invoked when host starts, and
                 // by requesting the TracerProvider from DI
                 // it ensures all instrumentations gets started.
-                this.serviceProvider.GetRequiredService<TracerProvider>();
+                this.serviceProvider.GetRequiredService<IEnumerable<TracerProvider>>();
```

Comment:
@cijothomas This is what I was saying on the SIG meeting today. Really super easy to support multiple TracerProviders registered through DI. Just asking the container here for all of the registrations will cause them all to spin up, and then they will all dispose when the container goes down (test now passes for that reason). 

If we don't want to make it easy, we could check the IEnumerable here and throw if more than one is found?

Reply:
While this solves the bug of not disposing a provider, I think this introduces another major issue:

```
var builder = new HostBuilder().ConfigureServices(services =>
            {
                services.AddOpenTelemetryTracing(builder =>
                {
                    builder.AddInstrumentation(() => testInstrumentation1);
                    builder.SetSampler<ParentBasedSampler>();
                });

                services.AddOpenTelemetryTracing(builder =>
                {
                    builder.AddInstrumentation(() => testInstrumentation2);
                    builder.SetSampler<ParentBasedSampler>();
                });

                // Which TracerProvider instance is being affected with the below line?
                services.AddSingleton<ParentBasedSampler>(new ParentBasedSampler(new AlwaysOffSampler()));
            });
```

Reply:
In that case, they both get the same one. The user has configured two TracerProviders. They differ by instrumentation but use the same sampler instance. I don't see a bug there, working as designed IMO,